src/Umbraco.Cms.Integrations.Search.Algolia/Services/AlgoliaIndexService.cs

```diff
@@ -43,7 +52,29 @@ public async Task<string> UpdateData(string name, Record record)
 
                 var index = client.InitIndex(name);
 
-                await index.PartialUpdateObjectAsync(record);
+                var obj = index.GetObjects<Record>(new[] { record.ObjectID }).FirstOrDefault();
+                if (obj != null)
+                    await index.PartialUpdateObjectAsync(record);
+                else
+                    await index.SaveObjectAsync(record, autoGenerateObjectId: false);
+                
+                return string.Empty;
```

Comment:
I see now why you are returning a string in that you check this in the caller to see if there has been an error.  But I don't see you return anything other than a string, so probably this could just be a void response.

Or, if you do want a result object, perhaps create a simple `Result` class, so it's a bit clearer what the response type is expected to contain.  Perhaps a simplified version of [this](https://github.com/umbraco/Umbraco.Marketplace/blob/main/src/PackagesRepository/PackagesRepository.Domain/Result.cs)?